kernel/src/process/posix_thread/mod.rs

```diff
@@ -242,20 +224,25 @@ impl PosixThread {
 
     /// Enqueues a thread-directed signal.
     ///
-    /// This method does not perform permission checks on user signals. Therefore, unless the
-    /// caller can ensure that there are no permission issues, this method should be used for
-    /// enqueue kernel signals or fault signals.
+    /// This method does not perform permission checks on user signals.
+    /// Therefore, unless the caller can ensure that there are no permission issues,
+    /// this method should be used to enqueue kernel signals or fault signals.
     pub fn enqueue_signal(&self, signal: Box<dyn Signal>) {
         let process = self.process();
         let sig_dispositions = process.sig_dispositions().lock();
         let sig_dispositions = sig_dispositions.lock();
 
-        let signum = signal.num();
-        if sig_dispositions.get(signum).will_ignore(signum) {
+        let will_ignore = sig_dispositions.will_ignore(signal.as_ref());
+        let blocked = self.has_signal_blocked(signal.num());
+        if will_ignore && !blocked {
             return;
         }
 
-        self.enqueue_signal_locked(signal, &sig_dispositions);
+        self.sig_queues.enqueue(signal);
+
+        if !blocked && !will_ignore {
```

Comment:
We should wake up some signalled wakers when changing the signal dispositions?

For example, modifying an ignored action into a signal handler for a signal that the current thread is blocking should wake up other non-blocking threads (because they can handle the signals immediately), right?

I cannot find the related code in the diff. Does it exist?